internal/sql/catalog/table.go

```diff
@@ -153,13 +153,24 @@ func (c *Catalog) createTable(stmt *ast.CreateTableStmt) error {
 		}
 	} else {
 		for _, col := range stmt.Cols {
-			tbl.Columns = append(tbl.Columns, &Column{
+			tc := &Column{
 				Name:      col.Colname,
 				Type:      *col.TypeName,
 				IsNotNull: col.IsNotNull,
 				IsArray:   col.IsArray,
 				Comment:   col.Comment,
-			})
+			}
+			if col.Vals != nil {
+				typeName := ast.TypeName{
+					Name: col.Colname,
```

Comment:
@kyleconroy I'm not 100% sure when this if triggers (for columns that use an implicit enum for the type, instead of a named one?) but this breaks with "type already exists" if it triggers twice for the same column name.

Thoughts on how to address? Prepending the table's name might be an easy-ish fix?

Reply:
https://github.com/kyleconroy/sqlc/pull/716